ddtrace/tracer/context_test.go

```diff
@@ -80,6 +80,42 @@ func TestStartSpanFromContext(t *testing.T) {
 	assert.Equal("/", got.Resource)
 }
 
+func TestStartSpanWithSpanLinks(t *testing.T) {
+	_, _, _, stop := startTestTracer(t)
+	defer stop()
+	spanLink := ddtrace.SpanLink{TraceID: 789, TraceIDHigh: 0, SpanID: 789, Attributes: map[string]string{"reason": "terminated_context", "context_headers": "datadog"}, Flags: 0}
+	var ctx ddtrace.SpanContext
+	ctx = &spanContext{spanID: 789, traceID: traceIDFrom64Bits(789), spanLinks: []ddtrace.SpanLink{spanLink}}
+
+	t.Run("create spancontext with links", func(t *testing.T) {
+		ctxLink, ok := ctx.(ddtrace.SpanContextWithLinks)
+		assert.True(t, ok)
+		assert.Equal(t, len(ctxLink.SpanLinks()), 1)
+		assert.Equal(t, ctxLink.SpanLinks()[0], spanLink)
+	})
+
+	t.Run("create span from spancontext with links", func(t *testing.T) {
+		var s ddtrace.Span
+		s, _ = StartSpanFromContext(
+			context.Background(),
+			"http.request",
+			WithSpanLinks([]ddtrace.SpanLink{spanLink}),
+			ChildOf(ctx),
+		)
+		assert.NotEmpty(t, s)
```

Comment:
Generally speaking this check isn't necessary (As in, I can't see a single flow through the codebase that would return a nil span from StartSpanFromContext, and we don't typically make this check in other tests); was there a particular reason you added it?

Reply:
No this is a good point. I can remove the check